### PR TITLE
Make keys required on python Table

### PR DESCRIPTION
**Pull request by patrick-schultz**
Scala `Table` still has `Option`al key for now. This PR just makes the user visible changes needed for 0.2 release.

### CODE REVIEW

File: python/hail/experimental/ldscore.py [hunk 1]
```diff
@@ -188,12 +188,17 @@ def ld_score(entry_expr,
     ht_scores = ht_scores.key_by('idx')
     ht_scores = ht_scores.rename({'f{:}'.format(i): col_idxs[i]
                                   for i in range(len(cols))})
+    ht_scores._force_count()
 
     ht = mt.select_rows(__locus=locus_expr).rows()
     ht = ht.add_index()
-    ht = ht.annotate(**ht_scores[ht.idx])
+    ht = ht.annotate(**ht_scores[ht.idx]) # dies here
```
Comment (catoverdrive) [inline python/hail/experimental/ldscore.py]: what dies here?

File: python/hail/matrixtable.py [hunk 2]
```diff
@@ -2984,7 +2984,6 @@ def from_rows_table(cls, table: Table) -> 'MatrixTable':
         -------
         :class:`.MatrixTable`
         """
-        hail.methods.misc.require_key(table, 'from_rows_table')
```
Comment (catoverdrive) [inline python/hail/matrixtable.py]: Can MatrixTables handle having an empty row key everywhere right now? The rvd partitioner stuff should work out fine, but do we make assumptions about the length of the row key elsewhere?

File: python/hail/experimental/ldscore.py (hunk 1 above)
Comment (catoverdrive) [reply python/hail/experimental/ldscore.py]: (the other comment that I appear to have forgotten to save---the `force_count()`s should probably be removed?)

Comment (patrick-schultz) [reply python/hail/experimental/ldscore.py]: Wow, I completely failed at cleaning up from a long debugging session.

File: python/hail/matrixtable.py (hunk 2 above)
Comment (patrick-schultz) [reply python/hail/matrixtable.py]: Good point. `require_key` only checks if the key is `None`, but maybe we should make it also check if the key is `[]`. I would expect this to be temporary, and that everything (or at least most things) should work fine with empty key.

**Pull request merged**